custom/icds_reports/tasks.py

```diff
@@ -308,29 +308,40 @@ def move_ucr_data_into_aggregation_tables(date=None, intervals=2, force_citus=Fa
 
+                res_ls_tasks = list()
+                res_ls_tasks.extend([
+                    icds_state_aggregation_task.si(
+                        state_id=state_id, date=calculation_date,
+                        func_name='_agg_ls_awc_mgt_form', force_citus=force_citus)
+                    for state_id in state_ids
+                ])
+                res_ls_tasks.extend([
+                    icds_state_aggregation_task.si(
+                        state_id=state_id, date=calculation_date,
+                        func_name='_agg_ls_vhnd_form', force_citus=force_citus)
+                    for state_id in state_ids
+                ])
+                res_ls_tasks.extend([
+                    icds_state_aggregation_task.si(
+                        state_id=state_id, date=calculation_date,
+                        func_name='_agg_beneficiary_form', force_citus=force_citus)
+                    for state_id in state_ids
+                ])
+
+                res_ls_task_results = [res_ls_task.delay() for res_ls_task in res_ls_tasks]
+                for res_ls_task_result in res_ls_task_results:
+                    res_ls_task_result.get(disable_sync_subtasks=False)
+
                 res_ccs.get(disable_sync_subtasks=False)
                 res_child.get(disable_sync_subtasks=False)
 
-                res_ls_tasks = list()
-                res_ls_tasks.extend([icds_state_aggregation_task.si(state_id=state_id, date=calculation_date,
-                                                                    func_name='_agg_ls_awc_mgt_form', force_citus=force_citus)
-                                     for state_id in state_ids
-                                     ])
-                res_ls_tasks.extend([icds_state_aggregation_task.si(state_id=state_id, date=calculation_date,
-                                                                    func_name='_agg_ls_vhnd_form', force_citus=force_citus)
-                                     for state_id in state_ids
-                                     ])
-                res_ls_tasks.extend([icds_state_aggregation_task.si(state_id=state_id, date=calculation_date,
-                                                                    func_name='_agg_beneficiary_form', force_citus=force_citus)
-                                     for state_id in state_ids
-                                     ])
-
-                res_ls_tasks.append(icds_aggregation_task.si(date=calculation_date, func_name='_agg_ls_table', force_citus=force_citus))
-
-                res_awc = chain(icds_aggregation_task.si(date=calculation_date, func_name='_agg_awc_table', force_citus=force_citus),
-                                *res_ls_tasks
-                                ).apply_async()
-
+                res_awc = icds_aggregation_task.si(
+                    date=calculation_date, func_name='_agg_awc_table', force_citus=force_citus
+                ).delay()
+                res_ls = icds_aggregation_task.si(
```

Comment:
I think this depends on agg_awc

Reply:
Can you show me where? The only tables I see referenced are three intermediate tables (they only depend on UCR) https://github.com/dimagi/commcare-hq/blob/20983d50bd9d33ae8120a3c2fb6b57b19ff09a0b/custom/icds_reports/utils/aggregation_helpers/monolith/agg_ls_data.py#L9, itself https://github.com/dimagi/commcare-hq/blob/20983d50bd9d33ae8120a3c2fb6b57b19ff09a0b/custom/icds_reports/utils/aggregation_helpers/monolith/agg_ls_data.py#L16, and `awc_location_local` https://github.com/dimagi/commcare-hq/blob/20983d50bd9d33ae8120a3c2fb6b57b19ff09a0b/custom/icds_reports/utils/aggregation_helpers/monolith/agg_ls_data.py#L100

Maybe this means you can also change your airflow to be more parallel? From what I can tell, I think you could even have an LS job totally separate from the AWC aggregations. (There's just a select for `awc_location_local` which should still be query-able even if the AWC agg is running

Reply:
This is fine to make the agg_ls independent to agg_awc. Initially when this code was merged it used to depened on agg_awc completion(not for through code but to keep the semantics of indicator right [see here](https://github.com/dimagi/commcare-hq/pull/22458/files#r235128250)).  But[ Later change](https://github.com/dimagi/commcare-hq/pull/22617/files#diff-77bde2e45c4a27b16ce4ef95e6e17a14R33) in the spec of LS dashboard invalidated that requirement and while making this later change I did not think of making it independent here.
@emord